engine/src/main/java/org/hibernate/search/engine/impl/ConfigContext.java

```diff
@@ -302,86 +253,57 @@ private void addFilterDef(FullTextFilterDef defAnn) {
 	}
 
 	public Map<String, AnalyzerReference> initLazyAnalyzerReferences(IndexManagerHolder indexesFactory) {
-		final Map<String, AnalyzerReference> initializedAnalyzers = new HashMap<>( analyzerDefs.size() );
-
-		for ( LuceneAnalyzerReference lazyAnalyzerReference : lazyLuceneAnalyzerReferences ) {
-			initLazyLuceneAnalyzer( initializedAnalyzers, lazyAnalyzerReference );
-		}
-
-		for ( RemoteAnalyzerReference remoteAnalyzerReference : lazyRemoteAnalyzerReferences ) {
-			initLazyRemoteAnalyzer( initializedAnalyzers, remoteAnalyzerReference, indexesFactory );
-		}
+		final Map<String, AnalyzerReference> referencesByName = new HashMap<>( analyzerDefs.size() );
 
-		// init default remote analyzers
-		initLazyRemoteAnalyzer( initializedAnalyzers, RemoteAnalyzerReference.DEFAULT, indexesFactory );
-		initLazyRemoteAnalyzer( initializedAnalyzers, RemoteAnalyzerReference.PASS_THROUGH, indexesFactory );
-
-		// initialize the remaining definitions
-		for ( Map.Entry<String, AnalyzerDef> entry : analyzerDefs.entrySet() ) {
-			if ( !initializedAnalyzers.containsKey( entry.getKey() ) ) {
-				final Analyzer analyzer = buildAnalyzer( entry.getValue() );
-				final AnalyzerReference reference = new LuceneAnalyzerReference( analyzer );
-				initializedAnalyzers.put( entry.getKey(), reference );
+		// Ensure every analyzer definition will be initialized
+		for ( String name : analyzerDefs.keySet() ) {
```

Comment:
As discussed on the phone, could you explain a bit more the sequence of operations. Typically here, it would be nice to explain that analyzers not used in the mapping are registered as Lucene analyzers.